pkg/apiserver/metrics/metrics.go

```diff
@@ -99,10 +100,17 @@ func InstrumentRouteFunc(verb, resource string, routeFunc restful.RouteFunction)
 		response.ResponseWriter = rw
 
 		routeFunc(request, response)
-		Monitor(&verb, &resource, utilnet.GetHTTPClient(request.Request), rw.Header().Get("Content-Type"), delegate.status, now)
+		Monitor(&verb, &resource, cleanUserAgent(utilnet.GetHTTPClient(request.Request)), rw.Header().Get("Content-Type"), delegate.status, now)
```

Comment:
You could make this depend on a flag, but if you try to turn these metrics into an debugging mechanism you're probably using them wrong.

The same goes for an extra label to tell it has been canonicalized.

If this is necessary to get this merged, ok, but I really think we shouldn't expose our users Prometheus servers to a huger number of (unnecessary) timeseries.